source/common/ssl/context_impl.h

```diff
@@ -37,8 +36,8 @@ struct SslStats {
   ALL_SSL_STATS(GENERATE_COUNTER_STRUCT, GENERATE_GAUGE_STRUCT, GENERATE_TIMER_STRUCT)
 };
 
-typedef CSmartPtr<SSL, SSL_free> SslConPtr;
-typedef CSmartPtr<X509, X509_free> X509Ptr;
+typedef bssl::UniquePtr<SSL> SslConPtr;
+typedef bssl::UniquePtr<X509> X509Ptr;
```

Comment:
Not really, it's strongly encouraged to do this in places like interfaces. But there are exceptions to every rule, and I think this is a valid one here (used in a few local places, working with some existing library and its conventions for pointers).